lib/active_merchant/billing/gateways/shift4.rb

```diff
@@ -0,0 +1,300 @@
+        'card_declined' => STANDARD_ERROR_CODE[:card_declined],
+        'processing_error' => STANDARD_ERROR_CODE[:processing_error],
+        'lost_or_stolen' => STANDARD_ERROR_CODE[:card_declined],
+        'suspected_fraud' => STANDARD_ERROR_CODE[:card_declined],
+        'expired_token' => STANDARD_ERROR_CODE[:card_declined]
+      }
+
+      def initialize(options = {})
+        requires!(options, :client_guid, :auth_token)
+        super
+      end
+
+      def access_token(options = {})
+        post = {}
+        add_credentials(post, options)
+        commit('accesstoken', post, request_headers(options))
+      end
+
+      def purchase(money, authorization, options = {})
+        post = {}
+        add_datetime(post, options)
+        api_options(post, options)
+        add_invoice(post, money, options)
+        add_clerk(post, options)
+        add_transaction(post, options)
+        add_card(post, authorization, options)
+
+        commit('sale', post, request_headers(options))
+      end
+
+      def authorize(money, card, options = {})
+        post = {}
+        add_datetime(post, options)
+        add_invoice(post, money, options)
+        add_clerk(post, options)
+        add_transaction(post, options)
+        add_card(post, card, options)
+        add_customer(post, options)
+
+        commit('authorization', post, request_headers(options))
+      end
+
+      def capture(money, authorization, options = {})
+        post = {}
+        add_datetime(post, options)
+        add_invoice(post, money, options)
+        add_clerk(post, options)
+        add_transaction(post, options)
+        add_card(post, authorization, options)
+
+        commit('capture', post, request_headers(options))
+      end
+
+      def refund(money, authorization, options = {})
+        post = {}
+        add_datetime(post, options)
+        api_options(post, options)
+        add_invoice(post, money, options)
+        add_clerk(post, options)
+        add_transaction(post, options)
+        add_device(post, options)
+        add_p2pe(post, options)
+        add_emv(post, options)
+        add_customer(post, options)
+        add_card(post, authorization, options)
+
+        commit('refund', post, request_headers(options))
+      end
+
+      def void(options = {})
+        post = {}
+        commit('invoice', post, request_headers(options))
+      end
+
+      def verify(credit_card, options = {})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(100, credit_card, options) }
+          r.process(:ignore_result) { void(r.authorization, options) }
+        end
+      end
+
+      private
+
+      def add_clerk(post, options)
+        post[:clerk] = {}
+        post[:clerk][:numericId] = options[:numeric_id]
+      end
+
+      def add_invoice(post, money, options)
+        post[:amount] = {}
+        post[:amount][:total] = money
+        post[:amount][:tax] = options[:amount][:tax] || 0.0
```

Comment:
Does this value need to be a `float` or an `int`? I see in unit test, we have a test value of `0` for this, but the remote test has a decimal value. 

If it's supposed to be a `float` we should use `int` for our test values and use `to_f` to type cast the value as we put it in the post body.